src/lib/DeviceFactory.ts

```diff
@@ -40,7 +40,9 @@ import WeatherForecast from './devices/WeatherForecast';
 import Window from './devices/Window';
 import WindowTilt from './devices/WindowTilt';
 
-const types: { [key in Types]: any } = {
+type ClassExtends<C> = { new (...args: any[]): C };
```

Comment:
Isn't it more like `ClassConstructor`? Or what does the extends mean here?

Reply:
Yes in fact it is "ClassConstructorOfOrExtended", but because if you "just want to have this class" then you do not need that because you can use the class itself as type, but this do not work for extended ones, so this special type is mainly used for Extended classes ... makes sense?

Reply:
Ok maybe a short jsdoc comment then